Задание 1-2

Divium-l opened the pull request:
1. Написана программа
2. Создана блок-схема алгоритма

Divium-l committed:
Задание 1-2

1. Написана программа
2. Создана блок-схема алгоритма

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -0,0 +1,67 @@
+
+			if ((isdigit(str[i])) || (str[i] == '.' && !isFloat))
+			{
+				if ((str[i] == '.' && isFloat)) isValid = false;
```
кажется, что это условие никогда не будет выполняться, недостижимый код

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -0,0 +1,67 @@
+			str.erase(0, 1);
+		}
+
+		for (int i = 0; i < str.length(); i++)
```
я бы лучше использовала цикл `while  (i < str.length() && isValid)) {}`

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -0,0 +1,67 @@
+#include <iostream>
+#include <string>
+#include <cctype>
+
```
`#include <cmath>` ?

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -0,0 +1,67 @@
+		isFloat = false;
+		isNegative = false;
+
+		cin >> str;
```
Почему-то Вы забыли пользователя попросить ввести число

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -0,0 +1,67 @@
+
+int main()
+{
+	double a = input();
```
я рада, что есть этот метод. 
задачи были написаны для `C#` и потом не отредактированы. В `C#` проверку осуществлять намнго проще.  Я пока всем не планирую давать задание с проверкой. Подумайте, этот метод можно и нужно изменить.. 
По задаче в целом. 
По всей програме хороший `naming`, но не при выборе `a` - `legth`?

Divium-l replied:
Не понял по поводу метода. Нужно как-то поменять функцию input()?

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -0,0 +1,67 @@
+	}
+
+	cout << endl << endl;
+	cout << "Face area = " << pow(a, 2) << endl;
```
вынесите все расчеты в отдельные функции, раз Вы это умеете.
лучше  в первых двух строках использовать `\n`, чтобы не прерывать поток исполнения
формула будет лучше читаться с пробелами `6 * pow(a,2)`
использование функций даст возможность при расчете полной площади использовать уже посчитанную площадь грани

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -0,0 +1,67 @@
+			if ((isdigit(str[i])) || (str[i] == '.' && !isFloat))
+			{
+				if ((str[i] == '.' && isFloat)) isValid = false;
+				if ((str[i] == '.' && !isFloat)) isFloat = true;
```
строку можно переписать `isFloat = str[i] == '.' ;`

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -0,0 +1,67 @@
+		}
+
+		if (!isValid) { cout << "Incorrect value! Please try again" << endl; }
+		else break;
```
старайтесь избегать `break`. Трудно понимать программу
старайтесь писать с правильными отсутпами, не экономя на форматировании

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -0,0 +1,67 @@
+
+		cin >> str;
+
+		if (str[0] == '-')
```
Я бы все-таки проверила, что у строки есть длина, и она больше нуля. Ну, уж после того, как "откусили" минус - точно проверять нужно - падает с грохотом(

MarinaPaley reviewed (changes requested):
Хочу похвалить за решение. Но, там есть над чем поработать!

Divium-l committed:
Задание 1-2 fix-1

1. Убран недостижимый код
2. Включена библиотека cmath
3. Попросили пользователя ввести число
4. Расчёты вынесены в отдельные функции
5. Убран break
6. Теперь программа не должна крашится при вводе “-”
7. Цикл for заменён на while
8. Маленькие исправления

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -35,33 +40,51 @@ double input() {
 	return stod(str);
+}
+
+double calcFaceArea(double length)
```
`(const double length)`

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -35,33 +40,51 @@ double input() {
+	return pow(length, 2);
+}
+
+double calcSurfaceArea(double length)
```
`(const double length)`

MarinaPaley commented on task-1-2/task-1-2.cpp:
```diff
@@ -35,33 +40,51 @@ double input() {
+	return 6 * calcFaceArea(length);
+}
+
+double calcVolume(double length)
```
`(const double length)`

MarinaPaley reviewed (changes requested):
Добавьте `const`, все остальное хорошо

Divium-l committed:
Задание 1-2 fix-2

1. Отправляем константы в функцию

MarinaPaley reviewed (approved):
Метод `input` имеет две разные ответственности: ввод пользовательский данных и `parse`. Так делать плохо, т.к. меод полцчается длинный и сложно воспринимаемый. Когда придет время, я всех группе буду объяснять как лучше делать "защиту дурака".
